okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/InstrumentedInterceptor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.okhttp;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import java.io.IOException;
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/**
+ * Records metrics about the response codes of http requests.
+ */
+public final class InstrumentedInterceptor implements Interceptor {
+
+    private final Meter informational;
+    private final Meter successful;
+    private final Meter redirection;
+    private final Meter clientError;
+    private final Meter serverError;
+    private final Meter other;
+
+    public InstrumentedInterceptor(MetricRegistry registry, String name) {
```

Comment:
Its not hooked up to the OkHttp client right now, how are you envisioning wiring in the metric registry?

Reply:
either plumbing it through the JaxRs.create(..) method or using the default shared metric registry: https://github.com/dropwizard/metrics/blob/4.0-development/metrics-core/src/main/java/com/codahale/metrics/SharedMetricRegistries.java